test/box-luatest/gh_6246_mvcc_different_select_behavior_test.lua

```diff
@@ -0,0 +1,103 @@
+local server = require('test.luatest_helpers.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all = function()
+    g.server = server:new{
+        alias   = 'default',
+        box_cfg = {memtx_use_mvcc_engine = true}
+    }
+    g.server:start()
+end
+
+g.after_all = function()
+    g.server:drop()
+end
+
+g.test_mvcc_different_select_behavior = function()
+    g.server:exec(function()
+        local t = require('luatest')
+        local fiber = require('fiber')
+
+        local s = box.schema.space.create('test')
+        s:create_index('primary')
+
+        local f = fiber.create(function()
+            fiber.self():set_joinable(true)
+            s:insert{1}
+        end)
+
+        t.assert_equals(s:select(), {})
+        t.assert_equals(s:select(1), {})
+        t.assert_equals(s:count(), 0)
+        box.begin()
+        local res1 = s:select()
+        local res2 = s:select(1)
+        local res3 = s:count()
+        box.commit()
+        t.assert_equals(res1, {})
+        t.assert_equals(res2, {})
+        t.assert_equals(res3, 0)
+
+        -- for RW transactions prepared statements are visible
```

Comment:
Nit: perhaps you could split different cases into different subtests, like you did here in https://github.com/tarantool/tarantool/commit/5c2dd8e3c6d6c2291529794bc30ef06463b99a0e?